web_search_with_and/static/src/js/search.js

```diff
@@ -0,0 +1,53 @@
+odoo.define('web_search_with_and', function (require) {
+    "use strict";
+
+    var SearchView = require('web.SearchView');
+    var Backbone = window.Backbone;
+
+    SearchView.include({
+        // Override the base method to detect a 'shift' event
+        select_completion: function (e, ui) {
+            if (e.shiftKey
+                && ui.item.facet.values
+                && ui.item.facet.values.length
+                && String(ui.item.facet.values[0].value).trim() !== "") {
+                // In case of an 'AND' search a new facet is added regarding of the previous facets
+                e.preventDefault();
+
+                this.query.add(ui.item.facet, {shiftKey: true});
+            } else {
+
+                return this._super.apply(this, arguments);
+            }
+        }
+    });
+
+    SearchView.SearchQuery.prototype = SearchView.SearchQuery.extend({
```

Comment:
I don't think so, this part was the real trick to migrate the module in v10.
I don't remember exactly the code but it was something like that : 
-  native v10 framework : method1 inherit method2
-  my goal : method1 inherit method_custom inherit method2 (place my code between existing inheritance)

This is how I usually deal with this problem, but if you have a better way to do it, please let me know :)